client/lib/client.dart

```diff
@@ -0,0 +1,717 @@
+Filter lt(String fieldPath, Object value) =>
+    _filter(fieldPath, Filter_Operator.LESS_THAN, value);
+
+/// Creates a field filter with the `>` operator.
+Filter gt(String fieldPath, Object value) =>
+    _filter(fieldPath, Filter_Operator.GREATER_THAN, value);
+
+Filter _filter(String fieldPath, Filter_Operator operator, Object value) {
+    ArgumentError.checkNotNull(fieldPath);
+    ArgumentError.checkNotNull(value);
+    var pathElements = fieldPath.split('.');
+    return Filter()
+        ..fieldPath = (FieldPath()..fieldName.addAll(pathElements))
+        ..operator = operator
+        ..value = packObject(value)
+        ..freeze();
+}
+
+/// A request to the server to post a command.
+class CommandRequest<M extends GeneratedMessage> {
+
+    final Client _client;
+    final Command _command;
+    final List<EventSubscription> _subscriptions = [];
+
+    CommandRequest._(this._client, M command) :
+            _command = _client._requests.command().create(command);
+
+    /// Creates an event subscription for events produced as a direct result of this command.
+    ///
+    /// Events down the line, i.e. events produced as the result of other messages which where
+    /// produced as the result of this command, do not match this subscription.
+    ///
+    /// Returns a broadcast stream of event messages.
+    ///
+    /// See `CommandRequest.observeEventsWithContexts(..)` to receive metadata, as well as the event
+    /// messages.
+    ///
+    ///
+    Stream<E> observeEvents<E extends GeneratedMessage>(E prototype) =>
+        _observeEvents(prototype).eventMessages;
+
+    /// Creates an event subscription for events produced as a direct result of this command.
+    ///
+    /// Events down the line, i.e. events produced as the result of other messages which where
+    /// produced as the result of this command, do not match this subscription.
+    ///
+    /// Returns a broadcast stream of events with their metadata.
+    ///
+    /// See `CommandRequest.observeEvents(..)` to unpacked event messages.
+    ///
+    Stream<Event> observeEventsWithContexts(GeneratedMessage prototype) =>
+        _observeEvents(prototype).events;
+
+    EventSubscription<E> _observeEvents<E extends GeneratedMessage>(E prototype) {
+        var subscription = _client.subscribeToEvents(prototype)
+            .where(all([eq('context.past_message', _commandAsOrigin())]))
+            .post();
+        _subscriptions.add(subscription);
+        return subscription;
+    }
+
+    Origin _commandAsOrigin() {
+        MessageId id = MessageId()
+            ..id = pack(_command.id)
+            ..typeUrl = _command.message.typeUrl;
+        return Origin()
+            ..message = id
+            ..actorContext = _command.context.actorContext;
+    }
+
+    /// Asynchronously sends this request to the server.
+    ///
+    /// Fails if there are no event subscription to monitor the command execution. If this is
```

Comment:
It looks like it should be `...if there is no event subscription...`.

Also, in Java we are able to observe changes to entity states. Maybe we could rephrase this piece so that we don't mention "event" subscription, but rather "a" subscription. It would help us in future, once entity state updates are supported — we won't have to modify this description. UPD. and the description below as well.

Reply:
We're able to observe entity state updates, just not as a result of a command. This is true for Java, JS, and this new Dart client.